### DISCUSSION TITLE
Outreachy 2022 applicants, start here!

### DISCUSSION THREAD

**Discussion by Angamanga** [ushahidi/platform#4213] [Outreachy and Google Summer of Code]
**Welcome to Ushahidi and the application-process for Outreachy!**

We are so excited you are here!! 🌻 🎉 💐  

There are a few steps you need to complete before starting working on an issue. These are:
1. Read through our [Code of Conduct](https://docs.ushahidi.com/platform-developer-documentation/code-of-conduct)
2. Get set up and say hello in [Gitter](https://gitter.im/ushahidi/gsoc-outreachy) to communicate, help and get help from the community.
3. Get your [developer-environment](https://docs.ushahidi.com/platform-developer-documentation/development-and-code/setup_alternatives) set up. We most often recommend using the Vagrant-setup, but XAMPP usually works fine too. Once you have set up your API, remember to setup the Platform-client as well! If you get stuck, reach out to the Gitter-channel to get help.

You also need to set up our bot-engine: https://github.com/ushahidi/ussd-engine. Follow the instructions in the readme to get setup! 

4. Explore the Ushahidi Platform and the different concepts like surveys (also called "forms" in some places) and posts. Try setting up a survey, a few categories, and add some posts. The user-documentation is found here: https://docs.ushahidi.com/ushahidi-platform-user-manual/

4. Once you have completed steps 1-4, send a reply to this discussion with screenshots of your setup in the browser and select an issue to work with.

**To keep in mind when starting working on issues:**
- Good issues to work with are found here: 
   - https://github.com/ushahidi/ussd-engine/labels/Outreachy2022
   - https://github.com/ushahidi/platform/labels/Outreachy2022

- Only work on one issue at a time. We are looking for the quality of the contributions rather than the number of contributions when evaluating the applications.

- Once you are assigned to an issue, we want you to submit a pull request within 3 days, even if it is not yet complete or a working solution. This is to make our contributors work "in the open" as is often done in Open Source and make sure no issues are not being worked on. If you don't show progress within 3 days, we will open it up for other students. Github has a good guide on how to submit pull requests from a fork: https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork
 
**Remember**
Ushahidi is an open-source community. The mentors are trying their best to reply to your questions, but there may be some delays. By helping other applicants out in Gitter, you are also helping the mentors. This *is* recognized by us and seen as something very positive in evaluating the applications. So if you see someone asking for help in the chat and know the answer, please help them out! Keep the tone nice and collaborative.

**Privacy info/personal info**
During the application-process, Ushahidi will store your Github-handle in a google-sheet. This is done only to make the application process as smooth as possible and to keep track of which issues are being worked on and the progress of each issue. The information stored will be deleted once the application-process is over.

**Comment by helen-ndip** [comment]
Hello @Angamanga  I have completed steps 1-4. Below are my screenshots. I would like to work on issue #4306 

1. Platform API
![image](https://user-images.githubusercontent.com/72542706/161386244-ac3e90eb-b1b9-4cc9-960b-bbb977eedc18.png)

2. Platform Client
![image](https://user-images.githubusercontent.com/72542706/161386955-bf9f3f00-ff50-4899-b907-504bebcc5856.png)


3. Bot-engine
![image](https://user-images.githubusercontent.com/72542706/161386341-a3f136ff-8a59-49b5-a59c-a6c8824361cf.png)

**Reply by helen-ndip** [reply]
While going through the installations, I noticed the following;
1. ```git clone``` command in the readme for bot-engine set up had a wrong public repo url ```git@github.com:ushahidi/ussd-engine.git``` instead of ```https://github.com/ushahidi/ussd-engine.git```.
2. In the installation of platform api on windows, the documentation says a password should not be added to the .htaccess file but without a password the ```composer migrate``` command fails. I needed to create a user and password to use for this configuration before it succeeded. 

Should I go ahead direct pull requests to fix them?

**Reply by yuktea** [reply]
@helen-ndip As for the first issue you pointer out, the URL is correct but it uses SSH authentication I believe

**Reply by helen-ndip** [reply]
> @helen-ndip As for the first issue you pointer out, the URL is correct but it uses SSH authentication I believe

I think so too reason why I could not access since I do not have the keys.

**Reply by Ifycode** [reply]
Hi @helen-ndip, url starting with `git@github.com` shows use of ssh connection, while url starting with `https://github.com` shows use of https connection. Sometime between August and October 2021, GitHub removed support for password authentication for when a Github user wants to perform a push or pull operation (when using https). Note: this doesn't affect cloning a repo. You can find time to read the **What you need to do today** section of this Github post: [Token authentication requirements for Git operations](https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/). If you ever have the push or pull issue I described above, you can always switch to using ssh. This part of the GitHub documentation will guide you: https://docs.github.com/en/authentication/connecting-to-github-with-ssh. You can always reach out if you need help with this at any point.

**Reply by helen-ndip** [reply]
> Hi @helen-ndip, url starting with `git@github.com` shows use of ssh connection, while url starting with `https://github.com` shows use of https connection. Sometime between August and October 2021, GitHub removed support for password authentication for when a Github user wants to perform a push or pull operation (when using https). Note: this doesn't affect cloning a repo. You can find time to read the **What you need today** section of this Github post: [Token authentication requirements for Git operations](https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/). If you ever have the push or pull issue I described above, you can always switch to using ssh. This part of the GitHub documentation will guide you: https://docs.github.com/en/authentication/connecting-to-github-with-ssh. You can always reach out if you need help with this at any point.

very correct. I did not set up the ssh keys.

**Reply by helen-ndip** [reply]
> Hi @helen-ndip, url starting with `git@github.com` shows use of ssh connection, while url starting with `https://github.com` shows use of https connection. Sometime between August and October 2021, GitHub removed support for password authentication for when a Github user wants to perform a push or pull operation (when using https). Note: this doesn't affect cloning a repo. You can find time to read the **What you need to do today** section of this Github post: [Token authentication requirements for Git operations](https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/). If you ever have the push or pull issue I described above, you can always switch to using ssh. This part of the GitHub documentation will guide you: https://docs.github.com/en/authentication/connecting-to-github-with-ssh. You can always reach out if you need help with this at any point.

@Ifycode  can we say issue 2 is a valid issue for many people though? Can I send a pull request on the docs for this?

**Reply by helen-ndip** [reply]
> > Hi @helen-ndip, url starting with `git@github.com` shows use of ssh connection, while url starting with `https://github.com` shows use of https connection. Sometime between August and October 2021, GitHub removed support for password authentication for when a Github user wants to perform a push or pull operation (when using https). Note: this doesn't affect cloning a repo. You can find time to read the **What you need to do today** section of this Github post: [Token authentication requirements for Git operations](https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/). If you ever have the push or pull issue I described above, you can always switch to using ssh. This part of the GitHub documentation will guide you: https://docs.github.com/en/authentication/connecting-to-github-with-ssh. You can always reach out if you need help with this at any point.
> 
> @Ifycode can we say issue 2 is a valid issue for many people though? Can I send a pull request on the docs for this?

Also docs says node version for platform client should be higher than 10 but in actual case it should be >=10 <13.

**Reply by Angamanga** [reply]
Hi @helen-ndip, good job on setting everything up. For issue 2 you can send a pr in the docs. Thank you for noticing 🙏

**Reply by helen-ndip** [reply]
Okay I will do just that thank you.

**Comment by yuktea** [comment]
Hey @Angamanga 
Done with Step 1-4! Can I begin working on Issue #4306 ?
![Screenshot (459)](https://user-images.githubusercontent.com/73362292/161434383-36308a2a-c1ab-476f-91f4-2286a3b9f0cd.png)
![Screenshot (460)](https://user-images.githubusercontent.com/73362292/161434390-4035abd6-a312-4b2b-bfd0-bd5071473f38.png)
![Screenshot (461)](https://user-images.githubusercontent.com/73362292/161434394-e8d221e8-14fe-489b-8cda-a4cd28d4129b.png)

**Comment by Angamanga** [comment]
@yuktea Good job setting up the environment. Please go ahead working on https://github.com/ushahidi/platform/issues/4306

**Reply by Angamanga** [reply]
@yuktea You need to comment in the issue before I can assign you to it.

**Comment by andrew21-mch** [comment]
hi @Angamanga, sorry for sending this late, i did not notice
I have been able to fully setup my development environment
Ussd-Engine
![bot](https://user-images.githubusercontent.com/58951422/161523676-943cb767-f19f-45c8-b231-31fa04dd70fa.png)
Ushahida Client
![client](https://user-images.githubusercontent.com/58951422/161523686-0fb1e5eb-575b-4e91-a190-6c1bc975a26c.png)
Ushahidi-Backend
![platform](https://user-images.githubusercontent.com/58951422/161523702-8baf632b-1ae9-43d4-973a-5bb407ff1327.png)

**Reply by Angamanga** [reply]
@andrew21-mch Good job 🎉

**Comment by andrew21-mch** [comment]
@Angamanga, i also noticed that while setting up the ussd-engine, some people might encounter an error when they take composer install

i recomend a
```composer update```
then 
```composer install```

**Reply by Angamanga** [reply]
@andrew21-mch Thank you for the recommendation, I'll update the readme 🙏

**Comment by Saumay1** [comment]
Hello @Angamanga!!
Completed the steps:
1. Platform API (Backend)
    
![Screenshot (280)](https://user-images.githubusercontent.com/46322717/161527953-dead38e0-3562-4f0e-9e30-c68bf8eb25d3.png)

2. Platform Client (Frontend)
    
![Screenshot (290)](https://user-images.githubusercontent.com/46322717/161528169-db2194e4-b171-4519-863e-ae4972fef5ef.png)

3. USSD Engine
    
![Screenshot (288)](https://user-images.githubusercontent.com/46322717/161528402-07d0cfb0-3601-4ec7-b943-db2aac69b400.png)

**Reply by Angamanga** [reply]
@Saumay1 Good job 🎉

**Comment by Saumay1** [comment]
Hi @Angamanga !
May I work on issue #4245.

**Reply by Angamanga** [reply]
@Saumay1 Yes, please go ahead!

**Comment by angeangoh** [comment]
![api](https://user-images.githubusercontent.com/100047512/162544360-2934a006-2df6-4b83-b668-658cff981e8b.png)
![client](https://user-images.githubusercontent.com/100047512/162544372-431ca07f-038b-483e-bb7f-4fac22f07fb7.png)
![uss](https://user-images.githubusercontent.com/100047512/162544383-7fb1ad61-ab43-408c-9a9f-2379661169b3.png)
hello @Angamanga  I set up my local environment and make an inspection on the features. Is there any available task for me because I don't know which one has already be taken by other interns? Anyway, I would like to work on this issue https://github.com/ushahidi/platform/issues/4384

**Comment by angeangoh** [comment]
Sorry to be late, I set up it for a while and forgot to come and post here.

**Comment by bkplaravel2018** [comment]
Hello @Angamanga
Done with Step 1-4! Can I start working on Issue https://github.com/ushahidi/platform/issues/4152 ? Or would you give me any good-first-issues which are not assigned to other?

Backend Setup:
![backend setup](https://user-images.githubusercontent.com/64826257/162573293-cfdd6d41-14c7-46aa-aa3c-a9a67d63b973.png)

Client Setup:
![client outcome](https://user-images.githubusercontent.com/64826257/162573391-b4436eb8-d34e-4296-971e-e56652828541.png)

BotMan-engine Setup:
![BotMan engine](https://user-images.githubusercontent.com/64826257/162573475-9fc88502-e428-4d51-a66b-16cb0ce562d1.png)

**Comment by ps-19** [comment]
Hi,
My name [Priyansh Singh](https://github.com/ps-19) . 

![WhatsApp Image 2022-04-14 at 6 19 31 PM](https://user-images.githubusercontent.com/63330165/163614968-ef364cb9-d8c3-41cd-bbeb-dfaca33d77c6.jpeg)
![WhatsApp Image 2022-04-14 at 6 19 48 PM](https://user-images.githubusercontent.com/63330165/163614985-fcfc2f49-0fd8-4bb7-ab41-c0740f7cc0f6.jpeg)
![WhatsApp Image 2022-04-14 at 6 19 56 PM](https://user-images.githubusercontent.com/63330165/163615010-afa3eb4c-58d6-4dda-8fbb-789e6382ffca.jpeg)
![WhatsApp Image 2022-04-14 at 6 20 03 PM](https://user-images.githubusercontent.com/63330165/163615028-82dbe1b3-7f0e-469e-b147-efbf9f0bb8eb.jpeg)
![WhatsApp Image 2022-04-14 at 6 20 15 PM](https://user-images.githubusercontent.com/63330165/163615049-2f26411f-c03f-4a07-bbd9-81281d78091d.jpeg)
![WhatsApp Image 2022-04-14 at 6 20 27 PM](https://user-images.githubusercontent.com/63330165/163615072-8d938ae1-a42a-45ff-8cd1-130e0c60d0f7.jpeg)
![WhatsApp Image 2022-04-14 at 6 20 36 PM](https://user-images.githubusercontent.com/63330165/163615093-d24d6e66-eeb5-4497-bc03-3fc7999ae9c6.jpeg)
![WhatsApp Image 2022-04-14 at 6 20 47 PM](https://user-images.githubusercontent.com/63330165/163615117-300da7fd-87fd-428e-b4cd-c373a43c20e8.jpeg)
![WhatsApp Image 2022-04-14 at 6 36 50 PM](https://user-images.githubusercontent.com/63330165/163615138-e194b5c8-912b-4fa3-8259-a1fb7c21a773.jpeg)
![WhatsApp Image 2022-04-14 at 6 55 23 PM](https://user-images.githubusercontent.com/63330165/163615167-722df550-b87a-4724-9947-c92d391955d1.jpeg)
![WhatsApp Image 2022-04-14 at 6 55 41 PM](https://user-images.githubusercontent.com/63330165/163615183-259707fe-79d6-4c69-b86a-83dc44372860.jpeg)
![Q-2](https://user-images.githubusercontent.com/63330165/163614913-53d98026-c104-4c72-a660-c758e70f413b.png)
![WhatsApp Image 2022-04-14 at 6 55 53 PM](https://user-images.githubusercontent.com/63330165/163615206-9a28761a-a4bc-4417-b145-e5930f5701b0.jpeg)
![WhatsApp Image 2022-04-14 at 6 56 05 PM](https://user-images.githubusercontent.com/63330165/163615227-27f0a3bc-bbb7-44dc-aa68-1749f8d24f4e.jpeg)
![WhatsApp Image 2022-04-14 at 6 56 16 PM](https://user-images.githubusercontent.com/63330165/163615250-84dc97ff-8107-4c03-a6c8-6923ea9db098.jpeg)
![WhatsApp Image 2022-04-14 at 6 56 26 PM](https://user-images.githubusercontent.com/63330165/163615271-c82a32dd-f649-4289-aa33-0417e715b21c.jpeg)
![WhatsApp Image 2022-04-14 at 6 56 41 PM](https://user-images.githubusercontent.com/63330165/163615291-b11e23db-a8bc-4475-958f-5d350f6e949d.jpeg)
![WhatsApp Image 2022-04-14 at 12 18 47 PM](https://user-images.githubusercontent.com/63330165/163615309-566c5415-dcbc-4f71-9923-337c806cbb2e.jpeg)

**Reply by ps-19** [reply]
**My relevant project based on Laravel framework:**
**[Posts](https://github.com/ps-19/Posts)** *Complete blogging website based on PHP. PHP Version of Blogger.com website, CRUD operations.*
**Read, Create, Edit, Delete the Post can be performed. Authentication is also available. A user can register on website, view blogs written by authenticated users.**
**Read, write, modify only by author option are provided. Images and videos insertion is also supported.**

**Comment by ps-19** [comment]
I have already worked on issue #4401 #4409 #4288.

PRs:
https://github.com/ushahidi/platform/pull/4408
https://github.com/ushahidi/platform/pull/4411
https://github.com/ushahidi/platform-client/pull/1779

**Reply by ps-19** [reply]
**My relevant project based on Laravel framework:**
**[Posts](https://github.com/ps-19/Posts)** *Complete blogging website based on PHP. PHP Version of Blogger.com website, CRUD operations.*
**Read, Create, Edit, Delete the Post can be performed. Authentication is also available. A user can register on website, view blogs written by authenticated users.**
**Read, write, modify only by author option are provided. Images and videos insertion is also supported.**

**Comment by ps-19** [comment]
@Angamanga mam please review

**Comment by ammarumer** [comment]
Hi @Angamanga I am completely new to open source and this first time that I am actually writing a comment on github after creating account which feels exciting.
My name is Ammar Umer and I am from Lahore,Pakistan albiet a small country in terms of technology and I'm outreachy applicant. For past 10 days I was completely sick with severe cough and fever and unfortunately due to which wasn't able to work on it and post earlier. And on top it took me 3 days to select project according to my preference and this was the one which I finally decided because it is close to my skillset as I have bit experience of PHP and Javascript.
I've followed your instructions and setup all the 3 directories using XAMP on windows and read the documentation as well.

Here are the setup screenshots
1. Api Backend
![image](https://user-images.githubusercontent.com/104008050/164010305-73c49f5c-3fb1-4d8b-b6b6-7cc2036487c8.png)

![image](https://user-images.githubusercontent.com/104008050/164010147-51f1ea82-56b5-4f89-9ed6-eef480b3807c.png)

2. Api Client
![image](https://user-images.githubusercontent.com/104008050/164010452-d2d86acb-3fd8-4d42-8d4a-f415715c6271.png)

![image](https://user-images.githubusercontent.com/104008050/164010538-5c00739c-5251-4083-8962-27e5b735cb75.png)

3. USSD Engine
There was one thing that I wasn't able to clone the ussd-engine directory with this URL git@github.com:ushahidi/ussd-engine.git
![image](https://user-images.githubusercontent.com/104008050/164011077-72c1acfa-b5d6-4f24-8a13-528f3b4579d7.png)


So I cloned with https URL and then setup the directory
![image](https://user-images.githubusercontent.com/104008050/164010727-4e454e06-8333-498c-8567-9319180fbc63.png)

I know I'm late to party but I would like to join and be part of your team and want to learn and contribute with hard work and dedication. Lastly I've browsed through the issues but mostly are taken for so I would like you to assign me any issue.

Thanks cheers

**Reply by ammarumer** [reply]
@Angamanga ma'am hope you are doing well. I would really appreciate if you could assign me an issue in which I could contribute plus learn as well. I have bit of experience in PHP so would like any task to make myself familiar with the API.
